Transition from ament_target_dependencies to target_link_libraries

fbe555 opened the pull request:
While using this repo with ROS Rolling Ridley, I have removed all uses of ament_target_dependencies, since it is now deprecated. I haven't tested this on anything but Rolling, and I'm no CMake wizard, but I thought I would create a PR to see if this might be useful. Let me know if there are any issues, and I'll be glad to try and address them as best I can.

Commit:
Transition from ament_target_dependencies to target_link_libraries

Since kilted, the use of ament_target_dependencies has been deprecated.
This commit replaces all uses of the macro with target_link_libraries as
suggested in the deprecation warning, and does some additional steps in
for the ethercat_interface package to ensure exports are done in the
"modern" target-based fashion.

fbe555 reviewed (commented):
I seem to have had something conveniently left over in my install dir, and not made a clean build. I cannot build this after cleaning. I will look into it tomorrow, and leave another comment when I have fixed it. Apologies for the premature PR.

fbe555 commented:
I have now pushed a commit with a fix, and the PR now builds on my system for Rolling. I had forgotten the project name subfolder in the install interface target include dir in the ethercat_interface CMakeLists.txt:
``` diff
@@ -35,7 +30,7 @@ add_library(

 target_include_directories(${PROJECT_NAME} PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-  $<INSTALL_INTERFACE:include>
+  $<INSTALL_INTERFACE:include/${PROJECT_NAME}>
   ${ETHERLAB_DIR}/include
 )
```
The PR should now be ready for running the repo test pipeline.

tpoignonec commented:
Hi, 
Sorry for the delay.
Since a lot of the users / maintainers still used humble until recently, the jazzy branch is lagging behind...
We'll merge stuff related to jazzy and later once the cumulative changes in PR #188 are merged into jazzy.

fbe555 commented:
Waiting to resolve conflicts until #195 and #196 are merged as well.

tpoignonec commented:
> Waiting to resolve conflicts until #195 and #196 are merged as well.

Should be good now, if you rebase you should be good to go.

Commit:
Update the way the yaml libraries are linked

When using the yaml_cpp_vendor package, the correct way to link agains
the yaml libraries is to first use:
  find_package(yaml_cpp_vendor REQUIRED)
  find_package(PkgConfig REQUIRED)
  pkg_check_modules(YAML_CPP REQUIRED yaml-cpp)
to find the concrete yaml vendor package, and then use the YAML_CPP_*
CMake variables to get the link targets, include libraries, and cmake
options. This commit updates the relevant cmake files to comply with
this pattern.

fbe555 commented:
I have now rebased the branch. I have also added a new commit, addressing the way the yaml library from yaml_cpp_vendor is linked, as this directly overlaps with this feature in that the target_link_libraries needs updating. I hope you don't mind the slight expansion of the PR. If so let me know and I'll try to split it.

fbe555 commented:
I'm going on christmas holiday starting today, but I'll keep an eye on this in case the PR fails the pipeline or review, but response times might be slightly longer than usual.

fbe555 commented:
ping

fbe555 commented:
Bumping this thread :)
